### PR TITLE
Warn user about data erasing after prepare step

### LINKED ISSUE #964: Prepared files removed for virtual/connect provision
If a file is created during the `prepare` step it will be removed during `push()` as it is not present in the local workdir and `rsync --delete` is used to sync files before executing testing. Reproducible with the `/tests/run/worktree` test:

    cd tests/run/worktree/data/prepare
    tmt run -a provision -h virtual

We need to investigate the best approach as simply removing the `--delete` option is not possible because remote files clean up is expected for example when re-running tests using `tmt run discover --force execute --force`.

One possible solution could be to check for the presence of the `--force` option, the other one to only synchronize selected folders (which would also improve the performance). Let's gather ideas here.

### PR DESCRIPTION
**Pull request by adiosnb**
Closes #964

### CODE REVIEW

**Review by lukaszachy — APPROVED** [review]
+1 but lets wait on @psss if wording is 100% correct.

File: spec/plans/prepare.fmf [hunk 1]
```diff
@@ -19,6 +19,16 @@ description: |
     individual tests order ``70`` is used, for recommended
     packages it is ``75``.
 
+    .. note::
+
+        If you want to use the ``prepare`` step to generate data
+        files for the testing in the ``exeute`` step, you need to
```
Comment (psss) [inline spec/plans/prepare.fmf:25]:
```suggestion
        files needed for testing during the ``execute`` step,
```

File: spec/plans/prepare.fmf [hunk 2]
```diff
@@ -19,6 +19,16 @@ description: |
...
+        move or copy them into ``${TMT_TEST_DATA}`` directory. Only
```
Comment (psss) [inline spec/plans/prepare.fmf:26]:
```suggestion
        move or copy them into ``${TMT_PLAN_DATA}`` directory. Only
```

File: spec/plans/prepare.fmf [hunk 3]
```diff
@@ -19,6 +19,16 @@ description: |
...
+        files in this directory are pulled back to the local machine
```
Comment (psss) [inline spec/plans/prepare.fmf:27]:
```suggestion
        files in this directory are guaranteed to be preserved.
```

I would probably skip the rest of the paragraph as those are rather implementation details, not interesting for the user and hard to explain precisely/correctly in a short sentence.

**Review by psss — CHANGES_REQUESTED** [review]
Thanks for putting this together. Looks ok, I've found two typos and suggest to omit the unnecessary implementation details at the end. What do you think?

**Pull request merged**